core/types/blob_sidecar.go

```diff
@@ -53,3 +55,34 @@ func (s *BlobSidecar) SanityCheck(blockNumber *big.Int, blockHash common.Hash) e
 	}
 	return nil
 }
+
+func (s *BlobSidecar) MarshalJSON() ([]byte, error) {
+	fields := map[string]interface{}{
+		"blockHash":   s.BlockHash,
+		"blockNumber": hexutil.EncodeUint64(s.BlockNumber.Uint64()),
+		"txHash":      s.TxHash,
+		"txIndex":     hexutil.EncodeUint64(s.TxIndex),
+	}
+	fields["blobSidecar"] = s.BlobTxSidecar
+	return json.Marshal(fields)
+}
+
+func (s *BlobSidecar) UnmarshalJSON(input []byte) error {
+	type blobSidecar struct {
+		BlobSidecar BlobTxSidecar `json:"blobSidecar"`
+		BlockNumber *hexutil.Big  `json:"blockNumber"`
+		BlockHash   common.Hash   `json:"blockHash"`
+		TxIndex     *hexutil.Big  `json:"txIndex"`
+		TxHash      common.Hash   `json:"txHash"`
+	}
+	var blob blobSidecar
+	if err := json.Unmarshal(input, &blob); err != nil {
+		return err
+	}
+	s.BlobTxSidecar = blob.BlobSidecar
+	s.BlockNumber = blob.BlockNumber.ToInt()
```

Comment:
You can refer to below line to avoid `nil pointer` issue.
https://github.com/bnb-chain/bsc/blob/26a4d4fda656cc78436c1931aaea5dc3ed33eeeb/core/types/gen_header_json.go#L124
```go
h.Difficulty = (*big.Int)(dec.Difficulty)
if dec.Number == nil {
	return errors.New("missing required field 'number' for Header")
}
```

Reply:
resolved